### PR TITLE
Parser: Recover from using colon as path separator in imports

### LINKED ISSUE #103269: A `:` instead of a `::` in a path of a `use` item should be a recoverable parsing error
I tried this code:

```rust
use std::process:Command;
use std::fs:File;

fn main() { }
```

I expected to see this happen: I expected to see two diagnostic messages, telling me about the distinct parse errors on lines 1 and 2.

Instead, this happened: The first error was treated as non-recoverable, yielding this output ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=90aeabb0d8644475e2614849634206df)):

```
error: expected one of `::`, `;`, or `as`, found `:`
 --> src/main.rs:1:17
  |
1 | [use std::process:Command;](https://play.rust-lang.org/#)
  |                 ^ expected one of `::`, `;`, or `as`

error: could not compile `playground` due to previous error
```

----

I'm **assuming** recovering in the face of this parse error would be a relatively simple task, given that there is already the `:` present that should be a strong hint that someone meant to type `::`.  So I'm tagging this with labels indicating that its a good opportunity for someone who wants to acquaint themselves with the code base.

### PR DESCRIPTION
**Pull request by mucinoab**
I don't know if this is the right approach, any feedback is welcome. 

r? @compiler-errors

Fixes #103269

### CODE REVIEW

File: compiler/rustc_parse/src/parser/item.rs [hunk 1]
```diff
@@ -971,6 +971,25 @@ impl<'a> Parser<'a> {
...
+                        .span_suggestion_short(
+                            self.token.span,
+                            "use double colon",
+                            "::",
+                            Applicability::MachineApplicable,
+                        )
+                        .note_once("import paths are delimited using `::`")
+                        .emit();
+
+                    self.bump(); // Colon
```
Comment (compiler-errors) [inline compiler/rustc_parse/src/parser/item.rs:986]:
```suggestion
                while self.eat_noexpect(&token::Colon) {
                    self.struct_span_err(self.prev_token.span, "expected `::`, found `:`")
                        .span_suggestion_short(
                            self.prev_token.span,
                            "use double colon",
                            "::",
                            Applicability::MachineApplicable,
                        )
                        .note_once("import paths are delimited using `::`")
                        .emit();
```

**Comment by compiler-errors** [comment]
One nitpick, otherwise this looks good to me! Can you test if it works, and then I can approve? @mucinoab

File: compiler/rustc_parse/src/parser/item.rs (hunk 1 above)
Comment (mucinoab) [reply compiler/rustc_parse/src/parser/item.rs:986]: Done.

**Comment by mucinoab** [comment]
All ready. @compiler-errors

**Pull request merged**